### PR TITLE
[mlir][spirv] Restructure code in `SPIRVConversion.cpp`. NFC.

### PR DESCRIPTION
**Pull request by obtuseangleAZ**

### CODE REVIEW

File: mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp [hunk 1]
```diff
@@ -40,19 +40,21 @@
...
 using namespace mlir;
 
+namespace {
+
 //===----------------------------------------------------------------------===//
 // Utility functions
 //===----------------------------------------------------------------------===//
 
-static int getComputeVectorSize(int64_t size) {
+int getComputeVectorSize(int64_t size) {
```
Comment (kuhar) [inline mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp:49]: Please leave `static` on this function -- it's somewhat redundant but it makes it easy to know if the current function is private or not

File: mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp [hunk 2]
```diff
@@ -731,73 +715,134 @@ std::optional<Value> castToSourceType(const spirv::TargetEnv &targetEnv,
...
 }
 
 //===----------------------------------------------------------------------===//
 // func::FuncOp Conversion Patterns
 //===----------------------------------------------------------------------===//
 
-namespace {
 /// A pattern for rewriting function signature to convert arguments of functions
 /// to be of valid SPIR-V types.
 class FuncOpConversion final : public OpConversionPattern<func::FuncOp> {
```
Comment (kuhar) [inline mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp:848]:
```suggestion
struct FuncOpConversion final : OpConversionPattern<func::FuncOp> {
```
and no `public:` below. Also please inline the implementation of `matchAndRewrite`

File: mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp [hunk 3]
```diff
@@ -855,16 +899,6 @@ FuncOpConversion::matchAndRewrite(func::FuncOp funcOp, OpAdaptor adaptor,
...
-}
-
-//===----------------------------------------------------------------------===//
-// func::FuncOp Conversion Patterns
-//===----------------------------------------------------------------------===//
-
-namespace {
 /// A pattern for rewriting function signature to convert vector arguments of
 /// functions to be of valid types
 struct FuncOpVectorUnroll final : OpRewritePattern<func::FuncOp> {
```
Comment (kuhar) [inline mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp:904]: Same here

Comment (angelz913) [reply mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp:900]: Could you please clarify what needs to be changed? It looks like it is already in the form mentioned in the previous form, and `matchAndRewrite` is inlined.

Comment (kuhar) [reply mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp:900]: ah, ok 👍

File: mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp [hunk 4]
```diff
@@ -52,7 +54,7 @@ static int getComputeVectorSize(int64_t size) {
   return 1;
 }
 
-static std::optional<SmallVector<int64_t>> getTargetShape(VectorType vecType) {
+std::optional<SmallVector<int64_t>> getTargetShape(VectorType vecType) {
```
Comment (kuhar) [inline mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp:57]: I meant re-adding static to all these functions where static was dropped.

**Review by kuhar — APPROVED** [review]

**Pull request merged**